crates/utils/re_arrow_util/src/transform.rs

```diff
@@ -958,13 +1003,102 @@ mod test {
         );
     }
+
+    // Generic test for binary arrays.
+    fn impl_binary_test<T: OffsetSizeTrait>(mut builder: GenericByteBuilder<GenericBinaryType<T>>) {
+        builder.append_value(b"hello");
+        builder.append_value(b"world");
+        builder.append_null();
+        builder.append_value(b"");
+        builder.append_value([0x00, 0xFF, 0x42]);
+        let binary_array = builder.finish();
+
+        println!("Input:");
+        println!("{}", DisplayRB(binary_array.clone()));
+
+        let result = BinaryToListUInt8::new().transform(&binary_array).unwrap();
+
+        println!("Output:");
+        println!("{}", DisplayRB(result.clone()));
+
+        // Verify structure
+        assert_eq!(result.len(), 5);
+        assert!(!result.is_null(0));
+        assert!(!result.is_null(1));
+        assert!(result.is_null(2));
+        assert!(!result.is_null(3));
+        assert!(!result.is_null(4));
+
+        {
+            let list = result.value(0);
+            let uint8 = list
+                .as_any()
+                .downcast_ref::<arrow::array::UInt8Array>()
+                .unwrap();
+            assert_eq!(uint8.len(), 5);
+            assert_eq!(uint8.value(0) as char, 'h');
+            assert_eq!(uint8.value(1) as char, 'e');
+            assert_eq!(uint8.value(2) as char, 'l');
+            assert_eq!(uint8.value(3) as char, 'l');
+            assert_eq!(uint8.value(4) as char, 'o');
+        }
+
+        {
+            let list = result.value(1);
+            let uint8 = list
+                .as_any()
+                .downcast_ref::<arrow::array::UInt8Array>()
+                .unwrap();
+            assert_eq!(list.len(), 5);
+            assert_eq!(uint8.value(0) as char, 'w');
+            assert_eq!(uint8.value(1) as char, 'o');
+            assert_eq!(uint8.value(2) as char, 'r');
+            assert_eq!(uint8.value(3) as char, 'l');
+            assert_eq!(uint8.value(4) as char, 'd');
+        }
+
+        assert!(result.is_null(2));
+
+        {
+            let list = result.value(3);
+            let uint8 = list
+                .as_any()
+                .downcast_ref::<arrow::array::UInt8Array>()
+                .unwrap();
+            assert_eq!(uint8.len(), 0);
+        }
+
+        {
+            let list = result.value(4);
+            let uint8 = list
+                .as_any()
+                .downcast_ref::<arrow::array::UInt8Array>()
+                .unwrap();
+            assert_eq!(uint8.len(), 3);
+            assert_eq!(uint8.value(0), 0x00);
+            assert_eq!(uint8.value(1), 0xFF);
+            assert_eq!(uint8.value(2), 0x42);
+        }
+    }
+
+    #[test]
+    fn test_binary_to_list_uint8() {
+        let builder = arrow::array::BinaryBuilder::new();
+        impl_binary_test(builder);
+    }
+
+    #[test]
+    fn test_large_binary_to_list_uint8() {
+        let builder = arrow::array::LargeBinaryBuilder::new();
+        impl_binary_test(builder);
+    }
```

Comment:
It would be good to also test that a 2GiB+ `LargeBinaryArray` produces the correct overflow error